[Feature] Support Primary Key table Backup & Restore

srlch opened the pull request:
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #11879

## Problem Summary(Required) ：
[Feature] Support Primary Key table Backup & Restore(#11879)

Problem:
Currently, StarRocks does not support the backup and restore of Primary Key tables. This feature will support it.

Solution:
We reused the code of the primary-key clone and transformed it to adapt to the backup restore process.

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 1 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|0|2|00.00%|[625, 626]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|0|1|00.00%|[82]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

decster reviewed (commented):
if is_restore_task==true, just create dir, do not go through the whole make_snapshot process, seems easier.

srlch commented on be/src/storage/tablet.cpp:
```diff
@@ -526,8 +526,13 @@ void Tablet::list_versions(vector<Version>* versions) const {
 
-Status Tablet::capture_consistent_rowsets(const Version& spec_version, std::vector<RowsetSharedPtr>* rowsets) const {
+Status Tablet::capture_consistent_rowsets(const Version& spec_version, std::vector<RowsetSharedPtr>* rowsets,
+                                          bool ingnore) const {
```
have fixed it.

srlch committed:
[Feature] Support Primary Key table Backup & Restore(#11879)

Problem:
At present, StarRocks does not support the backup and restore of Primary Key tables. This feature will support it.

Solution:
We basically reused the code of the primary key clone and transformed it to adapt to the backup restore process.

wanpengfei-git commented:
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/task/SnapshotTask.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/backup/BackupJob.java|1|1|100.00%|[]|

wanpengfei-git commented:
run starrocks_admit_test
